app/models/concerns/view_statistics.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module ViewStatistics
+  extend ActiveSupport::Concern
+
+  included do
+    has_many :view_statistics,
+             as: :resource,
+             dependent: :destroy
+  end
+
+  def count_view!
+    today = view_statistics.find_or_initialize_by(date: Time.zone.today)
+    today.update(count: today.count + 1)
+  end
```

Comment:
Fun fact, ActiveRecord actually has a method for this (because, of course it does). So if you wanted to, you could write this method like this. 

```ruby
  def count_view!
    view_statistics
      .find_or_initialize_by(date: Time.zone.today)
      .increment(:count)
      .save
  end
```